### PR TITLE
scalar and scalar-derive: fix typos

### PR DESCRIPTION
**Pull request by webern**
**Issue number:**

https://github.com/bottlerocket-os/bottlerocket/pull/2559#discussion_r1038657167

**Description of changes:**

Fix a couple of typos and add an assert as a follow up to #2559

**Testing done:**

None. There are no code changes.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by etungsten — APPROVED** [review]

File: sources/models/scalar-derive/README.md [hunk 1]
```diff
@@ -174,6 +174,10 @@ impl Validate for MyWrapper {
         Ok(Self{ some_field: input.into() })
     }
 }
+
+let value = MyWrapper::new(WeirdType).unwrap();
+// This type can be compared with &str because it we specified `as_ref_str = true`.
```
Comment (stmcginnis) [inline sources/models/scalar-derive/README.md]: "because it we"?

Comment (webern) [reply sources/models/scalar-derive/README.md]: :cry:

**Review by zmrow — APPROVED** [review]
Provided the `it we` typo is fixed - this looks good!

:wolf:

**Comment by webern** [comment]
^ fix `because it we`

https://github.com/bottlerocket-os/bottlerocket/compare/bf378ef18f0a135c320f050a6eb0c23d986f57b7..bc34c52d31dc77104f443a66d2bb07a21f84a2d9

**Review by stmcginnis — APPROVED** [review]
Looks good!

**Review by arnaldo2792 — APPROVED** [review]
Thanks! :tada:

**Review by bcressey — APPROVED** [review]

**Review by jpmcb — APPROVED** [review]

**Pull request merged**